# 🚑 MEDICAL INSURANCE COST PREDICTION USING MULTIPLE LINEAR REGRESSION

## ⚠️ Problem Statement

Medical insurance costs vary for individuals depending on several factors such as age, BMI, smoking habits, and family size. Understanding how these variables influence insurance charges is important for analyzing healthcare expenses. However, identifying these relationships manually from raw data can be difficult. Therefore, a data-driven approach is required to analyze the dataset and build a model that can estimate insurance costs based on relevant features.

## 🎯 Goal

The goal of this project is to develop a machine learning model that can predict medical insurance charges using demographic and health-related features.

## 🎯 Objectives
- Understand the structure and characteristics of the dataset
- Explore relationships between different variables and insurance charges
- Perform feature engineering to improve model understanding
- Build a Multiple Linear Regression model to predict insurance costs
- Evaluate model performance and interpret the results

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns
plt.style.use('ggplot')

In [2]:
import warnings
warnings.filterwarnings('ignore')

In [6]:
df=pd.read_csv("../dataset/medical_insurance.csv")
df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


## 📊 DATA UNDERSTANDING

In [7]:
df.shape

(2772, 7)

- The dataset contains 2772 rows and 7 columns, representing demographic and health-related attributes of individuals along with their medical insurance charges.

In [8]:
df.columns

Index(['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges'], dtype='object')

- The available features are age, sex, bmi, children, smoker, region, and charges, where charges is the target variable that the model will predict.

In [9]:
df.dtypes

age           int64
sex          object
bmi         float64
children      int64
smoker       object
region       object
charges     float64
dtype: object

- The dataset includes both numerical variables (age, bmi, children, charges) and categorical variables (sex, smoker, region). Categorical variables may require encoding before building the machine learning model.

In [11]:
df.describe()

,age,bmi,children,charges
count,2772.000000,2772.000000,2772.000000,2772.000000
mean,39.109668,30.701349,1.101732,13261.369959
std,14.081459,6.129449,1.214806,12151.768945
min,18.000000,15.960000,0.000000,1121.873900
25%,26.000000,26.220000,0.000000,4687.797000
50%,39.000000,30.447500,1.000000,9333.014350
75%,51.000000,34.770000,2.000000,16577.779500
max,64.000000,53.130000,5.000000,63770.428010


- From the statistical summary, the average age is around 39 years, the average BMI is approximately 30.7, and the average insurance charge is about 13,261. The insurance charges range from 1,121 to 63,770, indicating that some individuals incur significantly higher medical expenses.

## 🔍 HANDLING MISSING VALUES AND DUPLICATE VALUES

In [14]:
df.isnull().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

In [16]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
2767     True
2768     True
2769     True
2770     True
2771     True
Length: 2772, dtype: bool

In [17]:
df.duplicated().sum()

np.int64(1435)

In [18]:
(df.duplicated().sum() / len(df)) * 100

np.float64(51.76767676767676)

- The dataset was examined for duplicate records to ensure data quality and reliability. It was found that 50% of dataset contained duplicate rows, indicating that some observations were repeated.

In [19]:
df.drop_duplicates(inplace=True,keep="first")

In [20]:
df.duplicated().sum()

np.int64(0)

- Duplicate entries can introduce bias in analysis and machine learning models because the same data point may be counted multiple times. Therefore, these duplicate rows were removed so that each record represents a unique observation.

- After removing duplicates, the dataset now contains only distinct records, providing a cleaner and more reliable dataset for further analysis and modeling.